apache/superset
# feat(dashboard): Dashboard-Native Filters

### SUMMARY


Native filters is currently **not** feature-complete, but I am opening this PR anyway. We want to validate community acceptance of the core approaches before building the rest on top. This is behind a feature flag, so no need to worry about pushing out unpolished features.

Future enhancements will come as separate PRs. Because of this, some filter controls don't do anything yet.

Changes included in this PR:

- dashboard metadata filter schema
- dashboard-native filters can be created, edited, and deleted
- charts respond to filters when the user sets values
- filters can be scoped to specific dashboard panels
- ~upgrading react-redux to use ✨hooks✨~

Changes to be implemented in follow-up PRs:

- adjust styling to match design spec
- better filter delete UX
- filter indicator compatibility
- date filters
- cascading filters
- error boundaries
- non-instant filtering
- default values
- ... ?

This depends on a corresponding change in superset-ui

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


new filter bar:
<img width="1312" alt="Screen Shot 2020-11-25 at 12 21 16 PM" src="https://user-images.githubusercontent.com/1858430/100279424-5b75c780-2f1b-11eb-8ddc-0223ab6c13ac.png">

with filter bar closed:
<img width="1252" alt="Screen Shot 2020-11-25 at 12 21 46 PM" src="https://user-images.githubusercontent.com/1858430/100279437-629cd580-2f1b-11eb-801d-8b760adc08f8.png">

selecting a value:
<img width="269" alt="Screen Shot 2020-11-25 at 12 23 37 PM" src="https://user-images.githubusercontent.com/1858430/100279455-692b4d00-2f1b-11eb-9242-8d6b07374d8e.png">

charts filtered:
<img width="1680" alt="Screen Shot 2020-11-25 at 12 23 54 PM" src="https://user-images.githubusercontent.com/1858430/100279471-747e7880-2f1b-11eb-9a6e-ba3524de4d45.png">

filter configuration:
<img width="1676" alt="Screen Shot 2020-11-25 at 12 24 08 PM" src="https://user-images.githubusercontent.com/1858430/100279324-2b2e2900-2f1b-11eb-9e9d-0de547f71792.png">

### TEST PLAN


- (until the superset-ui PR is merged) check out the `villebro/filter-component` branch on superset-ui
- ensure that everything still works as expected with the feature flag off
- Turn on the `DASHBOARD_NATIVE_FILTERS` feature flag
- go to a dashboard
- use the filter bar to add some filters
- apply filter values
- edit filters, delete filters

Also, unit and e2e tests

### ADDITIONAL INFORMATION
